src/clusterfuzz/_internal/bot/tasks/utasks/variant_task.py

```diff
@@ -206,25 +206,36 @@ def handle_build_setup_error(output):
 
 def utask_postprocess(output):
   """Handle the output from utask_main."""
-  if output.testcase and environment.is_engine_fuzzer_job(
-      output.testcase.job_type):
-    # Remove put() method to avoid updates. DO NOT REMOVE THIS.
-    output.testcase.put = lambda: None
-
   if output.error_type is not None:
     uworker_handle_errors.handle(output, HANDLED_ERRORS)
     return
 
+  testcase = data_handler.get_testcase_by_id(output.uworker_input.testcase_id)
+  if environment.is_engine_fuzzer_job(output.uworker_input.job_type):
+    # Remove put() method to avoid updates. DO NOT REMOVE THIS.
+    testcase.put = lambda: None
```

Comment:
@jonathanmetzman remind me why do we need this again?

Reply:
Because variant task very hackily clones testcases and changes the engine in order to get crashes to reproduce, but we don't want the testcase to be updated with the fake engine. I think this is gross but it existed before my refactor.